hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/file/AvroResolver.java

```diff
@@ -37,7 +37,9 @@ public final class AvroResolver {
                     Schema.Type.LONG, QueryDataType.BIGINT,
                     Schema.Type.FLOAT, QueryDataType.REAL,
                     Schema.Type.DOUBLE, QueryDataType.DOUBLE,
-                    Schema.Type.STRING, QueryDataType.VARCHAR
+                    Schema.Type.STRING, QueryDataType.VARCHAR,
```

Comment:
do we have any tests for SQL File MAPPING with Avro?

Reply:
We have `com.hazelcast.jet.sql.impl.connector.file.SqlAvroTest`, but the schema types that are exclusively supported by file mappings are not tested. They are not expected to work currently due to the lack of serialization support for Avro records. The scope of HZ-3690 can be extended from `GenericRecord` to `GenericContainer` to support `ENUM`, `ARRAY` and `FIXED` in addition to `RECORD`. `MAP` and `BYTES` do not have special Avro containers and correspond to Java `Map` and `byte[]` respectively, so they might even be supported now.

Reply:
currently SQL cannot reasonably work with ENUM, ARRAY, MAP, FIXED, BYTES due to lack of syntax. Supporting them now by mapping to OBJECT will make it harder to introduce proper support later. The only things you can to with OBJECT is to convert it to string or return native Java class to the client (provided that the client is able to deserialize it)

In Pipeline API this is less of a problem as Jet pipeline is able to interact with GenericContainer objects directly. But for that we might also lack serialization support so sending such object between members might not work.

Reply:
table in https://docs.hazelcast.com/hazelcast/5.3/sql/mapping-to-a-file-system#avro says that we map everything else to `OBJECT` so your proposal is in line with that and would actually fix above mentioned types and we could support them the same way for Avro in Kafka.
It will be still impossible to insert to such mapping with OBJECT, but that is already known limitation.

Reply:
BTW, similar table in https://docs.hazelcast.com/hazelcast/5.3/sql/mapping-to-kafka#avro-messages says that everything else is mapped to `VARCHAR`. We should make this consistent.

Reply:
The first table is Avro → SQL, whereas the second one is SQL → Avro. I don't see any consistency problem.

Reply:
true, I did not notice that